### PR TITLE
TSK-1032: add confirmation dialog for projects, fix sprint deleting and allow deleting for Owner or creator only

### PR DESCRIPTION
**Pull request by ThetaDR**
# Contribution checklist

## Brief description
https://front.hc.engineering/workbench/platform/tracker/TSK-1032

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: plugins/tracker-resources/src/index.ts [hunk 1]
```diff
@@ -230,7 +230,20 @@ async function deleteProject (project: Project | undefined): Promise<void> {
         }
       )
     } else {
-      await removeProject(project)
+      showPopup(
```
Comment (haiodo) [inline plugins/tracker-resources/src/index.ts:232]: Why not use generic method to ask for confirmation and delete? Let's remove specialised one. This should be generic for any space. As well it should trigger delete of all child objects in space.

File: plugins/tracker-assets/lang/en.json [hunk 2]
```diff
@@ -181,6 +181,7 @@
     "EditProject": "Edit project",
     "DeleteProject": "Delete project",
     "DeleteProjectName": "Delete project {name}?",
+    "DeleteProjectConfirm": "Do you want to delete this project?",
```
Comment (haiodo) [inline plugins/tracker-assets/lang/en.json]: To archive?

File: plugins/tracker-resources/src/components/myissues/MyIssues.svelte [hunk 3]
```diff
@@ -54,7 +54,7 @@
   }
 
   function getQuery (mode: string, queries: { [key: string]: DocumentQuery<Issue> }) {
-    return queries[mode]
+    return { ...queries[mode], '$lookup.space.archived': false }
```
Comment (haiodo) [inline plugins/tracker-resources/src/components/myissues/MyIssues.svelte:57]: I think we need view option here.

**Review by haiodo — CHANGES_REQUESTED** [review]
We should have it named archive, and we need unarchive. Please check existing actions for them.

**Review by haiodo — APPROVED** [review]

File: plugins/tracker-resources/src/components/myissues/MyIssues.svelte (hunk 3 above)
Comment (ThetaDR) [reply plugins/tracker-resources/src/components/myissues/MyIssues.svelte:57]: Discussed, will be done in another PR

**Pull request merged**